util/eventValidation.js

```diff
@@ -242,111 +255,115 @@ async function validate(event) {
-                        logger.error(`Unknown option ${value} in ${key}"`);
-                    }
-                    break;
-                }
-                case "sendViolatedEventsTo": {
-                    if (value !== "procErrors") {
-                        logger.error(`Unknown option ${value} in ${key}"`);
-                    }
-                    break;
-                }
+    const violationsByType = new Set(validationErrors.map(err => err.type));
+    // eslint-disable-next-line no-restricted-syntax
+    for (const [key, val] of Object.entries(mergedTpConfig)) {
+      // To have compatibility for config-backend, spread-sheet plugin and postman collection
+      // We are making everything to lower string and doing string comparision.
+      const value = val?.toString()?.toLowerCase();
+      // eslint-disable-next-line default-case
+      switch (key) {
+        case "allowUnplannedEvents": {
+          const exists = violationsByType.has(violationTypes.UnplannedEvent);
+          if (value === "false" && exists) {
+            dropEvent = true;
+            violationType = violationTypes.UnplannedEvent;
+            break;
+          }
+          if (!(value === "true" || value === "false")) {
+            logger.error(`Unknown option ${value} in ${key}"`);
+          }
+          break;
+        }
+        case "unplannedProperties": {
+          const exists = violationsByType.has(
+            violationTypes.AdditionalProperties
+          );
+          if (value === "drop" && exists) {
+            dropEvent = true;
+            violationType = violationTypes.AdditionalProperties;
+            break;
+          }
+          if (!(value === "forward" || value === "drop")) {
+            logger.error(`Unknown option ${value} in ${key}"`);
+          }
+          break;
+        }
+        case "anyOtherViolation": {
+          const exists1 = violationsByType.has(violationTypes.UnknownViolation);
+          const exists2 = violationsByType.has(violationTypes.DatatypeMismatch);
+          const exists3 = violationsByType.has(violationTypes.RequiredMissing);
+          if (value === "drop" && (exists1 || exists2 || exists3)) {
+            if (exists1) {
+              violationType = violationTypes.UnknownViolation;
+            } else if (exists2) {
+              violationType = violationTypes.DatatypeMismatch;
+            } else {
+              violationType = violationTypes.RequiredMissing;
             }
+            dropEvent = true;
+            break;
+          }
+          if (!(value === "forward" || value === "drop")) {
+            logger.error(`Unknown option ${value} in ${key}"`);
+          }
+          break;
         }
-
-        return {
-            dropEvent: dropEvent,
-            violationType: violationType,
-            validationErrors: validationErrors
-        };
-    } catch (error) {
-        throw error;
+        case "sendViolatedEventsTo": {
+          if (value !== "procerrors") {
```

Comment:
Was this a typo earlier or now a breaking change?

Reply:
This is used only for logging purpose. Not a breaking change now or before.
Missed formatting to lowercase in last edit.